src/utils/filter.tsx

```diff
@@ -0,0 +1,65 @@
+import { Person } from '../types';
+import { AvailableFilters, FilterTypes, OrderType, SortOptions } from './enums';
+
+export const filter = (
+  selectedFilter: URLSearchParams,
+  peopleList: Person[],
+): Person[] => {
+  let preparedPeople = peopleList;
+  const sexTypes = selectedFilter.get(AvailableFilters.Sex);
+  const queryFilter = selectedFilter.get(AvailableFilters.Query);
+  const centuryFilter = selectedFilter.getAll(AvailableFilters.Centuries);
+  const sort = selectedFilter.get(AvailableFilters.Sort);
+  const order = selectedFilter.get(AvailableFilters.Order);
+
+  if (sexTypes && sexTypes !== FilterTypes.All) {
+    preparedPeople = preparedPeople.filter(person => person.sex === sexTypes);
+  }
+
+  if (centuryFilter.length) {
+    preparedPeople = preparedPeople.filter(person => {
+      const century = Math.ceil(person.born / 100).toString();
+
+      return centuryFilter.includes(century);
+    });
+  }
+
+  if (queryFilter) {
+    const lowerCaseQuery = queryFilter.toLowerCase();
+
+    preparedPeople = preparedPeople.filter(
+      person =>
+        person.name.toLowerCase().includes(lowerCaseQuery) ||
+        (person.fatherName &&
+          person.fatherName.toLowerCase().includes(lowerCaseQuery)) ||
+        (person.motherName &&
+          person.motherName.toLowerCase().includes(lowerCaseQuery)),
+    );
```

Comment:
i think u can modify it. Something like:
```suggestion
   preparedPeople = preparedPeople.filter(person => {
    const { name, fatherName = '', motherName = '' } = person;
    
    return [name, fatherName, motherName].some(field =>
      field.toLowerCase().includes(lowerCaseQuery)
    );
  });
```
it will be more readable

Reply:
done